ctx/ctx.go

```diff
@@ -0,0 +1,143 @@
+/*
+Current request context data.
+
+`kong.ctx.shared`:
+A table that has the lifetime of the current request and is shared between
+all plugins. It can be used to share data between several plugins in a given
+request.
+
+Since only relevant in the context of a request, this table cannot be
+accessed from the top-level chunk of Lua modules. Instead, it can only be
+accessed in request phases, which are represented by the `rewrite`,
+`access`, `header_filter`, `body_filter`, `log`, and `preread` phases of
+the plugin interfaces. Accessing this table in those functions (and their
+callees) is fine.
```

Comment:
is this paragraph relevant?  in Go plugins there's no "top-level" code.  maybe the equivalent would be the `New()` function (should we have other "initalization" or "config loading" hooks?)